jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/MavenWebInfConfiguration.java

```diff
@@ -52,14 +52,13 @@ public void configure(WebAppContext context) throws Exception
         MavenWebAppContext jwac = (MavenWebAppContext)context;
 
         //put the classes dir and all dependencies into the classpath
-        if (jwac.getClassPathFiles() != null && context.getClassLoader() instanceof WebAppClassLoader)
+        if (jwac.getClassPathUris() != null && context.getClassLoader() instanceof WebAppClassLoader loader)
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("Setting up classpath ...");
-            WebAppClassLoader loader = (WebAppClassLoader)context.getClassLoader();
-            for (File classpath : jwac.getClassPathFiles())
+            for (URI uri : jwac.getClassPathUris())
             {
-                loader.addClassPath(classpath.getCanonicalPath());
+                loader.addClassPath(uri.toASCIIString());
```

Comment:
It's probably worth writing a test that uses a `URI` pointing to a file that has a Swedish "å" character in its name to check that we can handle internationally-named jar files.

Reply:
We have those tests already.
And the use of `uri.toASCIIString()` ensure they are properly URI+UTF-8 encoded (if the input string hasn't encoded them).